protocol/messenger_mailserver.go

```diff
@@ -379,6 +379,40 @@ func (m *Messenger) RequestAllHistoricMessages(forceFetchingBackup, withRetries
 	return allResponses, nil
 }
 
+const missingMessageCheckPeriod = 30 * time.Second
+
+func (m *Messenger) checkForMissingMessages() {
+	go func() {
+		t := time.NewTimer(missingMessageCheckPeriod)
+		defer t.Stop()
+
+		for {
+			select {
+			case <-m.quit:
+				return
+
+			case <-t.C:
+				filters := m.transport.Filters()
```

Comment:
Would be good if this can be moved into a function under wakuv2/missing_messages.go aso that it can be  used for other scenarios such as disconnection and reconnection. 
If the API can take a time-range as an argument , it can be invoked whenever node comes back online as well.

Reply:
That would be interesting 
The part I see somewhat complicated is how to obtain the active mailserver at the waku layer, since they can be either a storenode from the fleet, or a storenode assigned to a community, and communities are a concept that the waku layer should probably not know about. :thinking:

Reply:
Agreed that community is something that should not be brought down to waku layer. But when a storenode from community needs to be used the caller can pass an optional WithStoreNode flag like we do with peer selection. 
In other cases, peer-selection using shard/pubsubtopic should help figure out which storenode to use. 
This should be fine, unless i am missing something.